quadriga/src/main/java/edu/asu/diging/quadriga/api/v1/MapGraphToTripleController.java

```diff
@@ -0,0 +1,119 @@
+
+import edu.asu.diging.quadriga.api.v1.model.PatternMapping;
+import edu.asu.diging.quadriga.api.v1.model.PatternMappingList;
+import edu.asu.diging.quadriga.core.model.EventGraph;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.jobs.Job;
+import edu.asu.diging.quadriga.core.service.AsyncPatternProcessor;
+import edu.asu.diging.quadriga.core.service.EventGraphService;
+import edu.asu.diging.quadriga.core.service.JobManager;
+
+@Controller
+public class MapGraphToTripleController {
+
+    @Value("${quadriga_base_url}")
+    private String quadrigaBaseUri;
+    
+    @Value("${quadriga_job_status_api}")
+    private String quadrigaJobStatusUri;
+    
+    @Value("${quadriga_collection_page}")
+    private String quadrigaCollectionPageUri;
+    
+    @Autowired
+    private AsyncPatternProcessor asyncPatternProcessor;
+
+    @Autowired
+    private EventGraphService eventGraphService;
+
+    @Autowired
+    private JobManager jobManager;
+
+    @PostMapping(value = "/api/v1/collection/{collectionId}/network/map")
+    public ResponseEntity<List<JobPatternInfo>> mapPatternToTriples(@PathVariable String collectionId,
+            @RequestBody PatternMappingList patternMappingList) {
+
+        List<EventGraph> eventGraphs = eventGraphService.getEventGraphs(new ObjectId(collectionId));
+        if (eventGraphs == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        
+        List<JobPatternInfo> jobInfos = new ArrayList<>();
+        
+        MappedTripleGroup mappedTripleGroup = new MappedTripleGroup();
+        mappedTripleGroup.set_id(new ObjectId());
+        mappedTripleGroup.setCollectionId(new ObjectId(collectionId));
+        
+        String jobId = jobManager.createJob(collectionId, mappedTripleGroup.get_id().toString(), eventGraphs.size());
+        
+        List<Thread> transformationThreads = new ArrayList<>();
+        for (PatternMapping pattern : patternMappingList.getPatternMappings()) {
+            Thread transformationThread = new Thread(() -> {
+                asyncPatternProcessor.processPattern(jobId, collectionId, pattern, eventGraphs);
+                JobPatternInfo jobInfo = new JobPatternInfo();
+                jobInfo.setJobId(jobId);
+                jobInfo.setTrack(quadrigaBaseUri + quadrigaJobStatusUri + jobId + "/status");
+                jobInfo.setExplore(quadrigaBaseUri + quadrigaCollectionPageUri + collectionId);
+                jobInfos.add(jobInfo);
+            });
+            transformationThreads.add(transformationThread);
+            transformationThread.start();
+        }
+        for (Thread thread : transformationThreads) {
+            try {
+                thread.join();
+            } catch (InterruptedException e) {
+                System.out.println("Job thread was interrupted due to the exception:"+e);
+            }
+        }
+        return new ResponseEntity<>(jobInfos, HttpStatus.OK);
```

Comment:
There should be just one job for a transformation request. There might be multiple threads started to work on the individual patterns, but there should only be one job.